src/Debug.mo

```diff
@@ -9,7 +9,8 @@ import Prim "mo:⛔";
 module {
   /// Prints `text` to output stream.
   ///
-  /// NOTE: What this output stream is depends on your execution environment.
+  /// NOTE: The output is placed in the replica log. When running on mainnet, it
```

Comment:
@crusso Outside of the scope of this PR, but this brings up a few questions for me.

How difficult would it be for a canister to access it's own replica log, or to provide an API to do so?
How large is the replica log, and under what circumstances does it fill up/get cleared?

Reply:
Not possible, unless you redirect the debug_print call to your own implementation. In production, calls to `debug_print` is actually a no-op: https://internetcomputer.org/docs/current/references/ic-interface-spec#debugging_aids

@kentosugama I just noticed that the "do nothing" behavior in production is actually specified in the interface spec above. We can be more explicit here, and say the call is a no-op when running on the mainnet.